### DISCUSSION TITLE
translate from nginx.json

### DISCUSSION THREAD

**Discussion by TWART016** [NginxProxyManager/nginx-proxy-manager#1522] [Q&A]
Hi,

I want to use the tool vikunja through npm. In the docs there is an example for nginx configuration. How can I translate the config into the npm GUI?

https://vikunja.io/docs/full-docker-example/#example-with-nginx-as-proxy

```
server {
    listen 80;

    location / {
        proxy_pass http://frontend:80;
    }

    location ~* ^/(api|dav|\.well-known)/ {
        proxy_pass http://api:3456;
        client_max_body_size 20M;
    }
}
```

**Comment by chaptergy** [comment]
You should be able to add a normal proxy host targeting `frontend` on port `80`, then you can switch to the locations tab and add a custom location `~* ^/(api|dav|\.well-known)/` targeting `api` on port `3456`. Both should have http as the protocol. You can then click on the cog button on your newly added location and add the `client_max_body_size 20M;` line in the now displayed text area.

**Reply by TWART016** [reply]
I set these settings. But these lines are missing:


```
    location / {
        proxy_pass http://frontend:80;
    }
```

The login failed on the GUI:

```
Console:
user-settings.89ce70d7.js:10 Mixed Content: The page at 'https://MYDOMAIN/login' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://192.168.178.15:3456/api/v1/login'. This request has been blocked; the content must be served over HTTPS.
```

Log:
```
[26/Oct/2021:22:06:38 +0000] - 304 304 - GET https MYDOMAIN "/login" [Client PUBLIC-IP] [Length 0] [Gzip -] [Sent-to 192.168.178.15] "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.54 Safari/537.36" "-"
[26/Oct/2021:22:06:40 +0000] - 200 200 - GET https MYDOMAIN "/sw.js" [Client PUBLIC-IP] [Length 6845] [Gzip -] [Sent-to 192.168.178.15] "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.54 Safari/537.36" "https://MYDOMAIN/sw.$
```
https://github.com/go-vikunja/frontend/issues/28

**Reply by chaptergy** [reply]
Those lines are not missing, they are included in all proxy hosts by default. Your issue seems to be that your frontend expects the API to be at `http://192.168.178.15:3456/api`, which is simply not correct. The frontend has to connect to the API via `https://MYDOMAIN/api`. So you have to [tell your frontend where to find the correct api](https://vikunja.io/docs/install-frontend/#api-url-configuration-in-docker).

**Reply by TWART016** [reply]
I set the parameter in the compose
`VIKUNJA_API_URL: http://192.168.178.15:3456/api/v1/
`

If I change that to https I get the error:
```
user-settings.89ce70d7.js:10 Mixed Content: The page at 'https://MYDOMAIN/login' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://192.168.178.15:3456/api/v1/info'. This request has been blocked; the content must be served over HTTPS.
(anonymous)	@	user-settings.89ce70d7.js:10
El	@	user-settings.89ce70d7.js:10
bv	@	user-settings.89ce70d7.js:10
Sr.request	@	user-settings.89ce70d7.js:10
Sr.<computed>	@	user-settings.89ce70d7.js:10
(anonymous)	@	user-settings.89ce70d7.js:9
update	@	index.e4884bbb.js:16
(anonymous)	@	user-settings.89ce70d7.js:9
fe.dispatch	@	user-settings.89ce70d7.js:9
dispatch	@	user-settings.89ce70d7.js:9
beforeCreate	@	index.e4884bbb.js:8
jt	@	user-settings.89ce70d7.js:5
ke	@	user-settings.89ce70d7.js:5
Rf.e._init	@	user-settings.89ce70d7.js:5
l	@	user-settings.89ce70d7.js:5
Xh	@	user-settings.89ce70d7.js:5
init	@	user-settings.89ce70d7.js:5
c	@	user-settings.89ce70d7.js:5
u	@	user-settings.89ce70d7.js:5
(anonymous)	@	user-settings.89ce70d7.js:5
e._update	@	user-settings.89ce70d7.js:5
n	@	user-settings.89ce70d7.js:5
Ee.get	@	user-settings.89ce70d7.js:5
Ee	@	user-settings.89ce70d7.js:5
ff	@	user-settings.89ce70d7.js:5
F.$mount	@	user-settings.89ce70d7.js:5
(anonymous)	@	index.e4884bbb.js:504

user-settings.89ce70d7.js:9 Uncaught (in promise) Error: Network Error
    at xl (user-settings.89ce70d7.js:9)
    at XMLHttpRequest.o.onerror (user-settings.89ce70d7.js:10)
xl	@	user-settings.89ce70d7.js:9
o.onerror	@	user-settings.89ce70d7.js:10
Promise.then (async)		
beforeCreate	@	index.e4884bbb.js:8
jt	@	user-settings.89ce70d7.js:5
ke	@	user-settings.89ce70d7.js:5
Rf.e._init	@	user-settings.89ce70d7.js:5
l	@	user-settings.89ce70d7.js:5
Xh	@	user-settings.89ce70d7.js:5
init	@	user-settings.89ce70d7.js:5
c	@	user-settings.89ce70d7.js:5
u	@	user-settings.89ce70d7.js:5
(anonymous)	@	user-settings.89ce70d7.js:5
e._update	@	user-settings.89ce70d7.js:5
n	@	user-settings.89ce70d7.js:5
Ee.get	@	user-settings.89ce70d7.js:5
Ee	@	user-settings.89ce70d7.js:5
ff	@	user-settings.89ce70d7.js:5
F.$mount	@	user-settings.89ce70d7.js:5
(anonymous)	@	index.e4884bbb.js:504
```

**Reply by chaptergy** [reply]
You don't just have to change `http` to `https` but also `192.168.178.15:3456` to `MYDOMAIN`. So your variable is `VIKUNJA_API_URL: https://MYDOMAIN/api/v1/`. Not sure where the variable has to point to, it might possibly also be `VIKUNJA_API_URL: https://MYDOMAIN/api/` or `VIKUNJA_API_URL: https://MYDOMAIN/`. Also remember you need to docker-compose up again so the environment variables get updated.

**Reply by chaptergy** [reply]
It seems a member of the vikunja team already told you how they handle the environment variable and told you what to do in https://github.com/go-vikunja/frontend/issues/28#issuecomment-952563952:

> On the other hand, since you are now using a proxy you should be fine to remove it all together as long as you're accessing everything only through that proxy.

So just go ahead and remove the variable alltogether.

**Reply by TWART016** [reply]
After removing the URL I can login. Just the usage with the internal IP is now difficult.
